app/helpers/search_helper.rb

```diff
@@ -6,4 +6,32 @@ def results_order_notice_for_form(form)
       I18n.t('search.alphabetical_order_notice')
     end
   end
+
+  def firm_has_qualifications_or_accreditations?(firm)
+    return true if firm.adviser_qualification_ids.any? do |id|
+      qualification_or_accreditation_key(id, :qualification) != 'ignored'
+    end
+
+    return true if firm.adviser_accreditation_ids.any? do |id|
+      qualification_or_accreditation_key(id, :accreditation) != 'ignored'
+    end
```

Comment:
Can definitely see your point. I've added a few unit tests for this & the other method I added. They're a little unusual, but I think they're the best option in this case. lmk wyt